lib/install/management/appliance.go

```diff
@@ -1180,3 +1194,25 @@ func (d *Dispatcher) CheckServiceReady(ctx context.Context, conf *config.Virtual
 	}
 	return nil
 }
+
+// deleteVCHFolder will delete an empty VCH Folder
+func (d *Dispatcher) deleteVCHFolder() {
+	// only continue if VC and the VCH Folder is NOT the datacenter wide VM Folder
+	if d.isVC && d.session.VCHFolder != nil && d.session.VCHFolder.Reference() != d.session.VMFolder.Reference() {
```

Comment:
Do we need to protect against `d.session.VMFolder` being `nil`?

Reply:
_possibly_ -- if we make a change I'd prefer to fail `vic-machine` all together if the `VMFolder` can't be set -- that would happen [here](https://github.com/vmware/vic/blob/master/lib/install/validate/validator.go#L160) and probably require a change to session.

Reply:
@zjs for now I'd prefer not to make a change around VMFolder -- currently in `vic-machine` a similar panic possibility exists.  I'll open an issue around protecting against that possibility that we can address in a future sprint.

Reply:
Issue created: #7692